### DISCUSSION TITLE
Data Engineers

### DISCUSSION THREAD

**Discussion by arneschreuder** [COS-301/graduates#17] [General]
This is the discussion page for and with Data Engineers.

**Comment by arneschreuder** [comment]
https://docs.nestjs.com/recipes/prisma
https://www.prisma.io/studio
https://www.prisma.io/docs/concepts/components/prisma-migrate

**Comment by LarisaBothaTuks** [comment]
@arneschreuder. I am the data-engineer of the storage team. Inside the discussion of my first pull request you mentioned that I should "hide" the firebase config data. I must have misunderstood what that meant by moving them to the env file. Can you please provide some clarity on how to achieve this?

**Reply by arneschreuder** [reply]
Hi @LarisaBothaTuks. You need to put configuration keys and api tokens and stuff like that in the “environments/environment.whatever.ts”, .env can also work. These are two different strategies. 

So basically, it has to be in a configuration that does not get committed, but rather read off of environment variables or a config system of sort. NestJS has both of these covered. Its up to you to pick one. 

GitHub has the concept of environment secrets. These are equivalent to .env. You can work with ci/cd or delivery team to perhaps put firebase configs in there. That way they will be available during builds. This is what you need. Runtime configurations are to be considered as well. 

I suggest perhaps do some research on nestjs configuration and firebase configuration for open source repositories. There are many ways to tackle it. It is up to you to figure out.

**Reply by LarisaBothaTuks** [reply]
Thank you for pointing me in a direction I will study up on these strategies
